### PR TITLE
enable Werror in CI

### PR DESCRIPTION
**Pull request by BenjaminW3**
This enables -Werror for clang, gcc and MSVC in the CI.

### CODE REVIEW

**Comment by BenjaminW3** [comment]
Ready to merge!

File: test/analysis/headerCheck/src/main.cpp [hunk 1]
```diff
@@ -20,7 +20,16 @@
...
 #define BOOST_TEST_MODULE headerCheck
+
+#include <boost/predef.h>   // BOOST_COMP_CLANG
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
 #include <boost/test/unit_test.hpp>
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic pop
```
Comment (psychocoderHPC) [inline test/analysis/headerCheck/src/main.cpp]: Is there a need to disable all this pragma's. 
I know that unknown pragmas result in a warning but maybe we should disable the warning for unknown pragmas.

Comment (BenjaminW3) [inline test/analysis/headerCheck/src/main.cpp]: Most of the time the warning about unknown pragmas is not useful. However, in the case of alpaka that defines many macros containing pragmas, the warning most probably shows a bug within alpaka. Therefore I would not want to fully disable it.

Comment (psychocoderHPC) [inline test/analysis/headerCheck/src/main.cpp]: ok I am fine with it

Comment (BenjaminW3) [inline test/analysis/headerCheck/src/main.cpp]: Is there anything else to do before it can be merged?

Comment (psychocoderHPC) [inline test/analysis/headerCheck/src/main.cpp]: merged

**Pull request merged**